src/Illuminate/Database/Query/Builder.php

```diff
@@ -2017,6 +2046,52 @@ protected function stripTableForPluck($column)
         return is_null($column) ? $column : last(preg_split('~\.| ~', $column));
     }
 
+    /**
+     * Retrieve column values from rows represented as objects.
+     *
+     * @param  array  $queryResult
+     * @param  string $column
+     * @param  string $key
+     * @return \Illuminate\Support\Collection
+     */
+    protected function pluckFromObjectColumn($queryResult, $column, $key)
+    {
+        $results = [];
+
+        foreach ($queryResult as $row) {
+            if (is_null($key)) {
+                $results[] = $row->$column;
+            } else {
+                $results[$row->$key] = $row->$column;
+            }
+        }
```

Comment:
I would move the `is_null()` function call outside of the loop:
```php
if (is_null($key)) {
    foreach ($queryResult as $row) {
        $results[] = $row->$column;
    }
} else {
    foreach ($queryResult as $row) {
        $results[$row->$key] = $row->$column;
    }
}
```
It's a bit longer, but saving thousands of function calls is worth it, and the code is still readable, maybe even a bit better.